python/ray/experimental/channel/cpu_nccl_group.py

```diff
@@ -0,0 +1,157 @@
+import asyncio
+from collections import defaultdict
+from typing import Optional, Tuple, List
+from unittest import mock
+
+import torch
+
+import ray
+import ray.experimental.channel as ray_channel
+from ray.experimental.channel.gpu_communicator import TorchTensorAllocator, ReduceOp
+
+
+@ray.remote(num_cpus=0)
+class Barrier:
+    """
+    Barrier that blocks the given number of actors until all actors have
+    reached the barrier. This is used to mock out blocking NCCL ops.
+    """
+
+    def __init__(self, num_actors=2):
+        self.num_actors = num_actors
+        self.condition = asyncio.Condition()
+        # Buffer for the data that is "sent" between the actors, each entry is
+        # one p2p op.
+        self.data = {}
+        self.collective_data = defaultdict(list)
+        # Buffer for the number of actors seen, each entry is one p2p op.
+        self.num_actors_seen = defaultdict(int)
+
+    async def wait(self, op_id: int, data=None):
```

Comment:
Maybe rename to indicate this is to replace p2p ops.

Reply:
Fixed